core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -732,6 +731,8 @@ private[spark] class TaskSetManager(
         return
       }
     }
+    sched.dagScheduler.taskResubmit(tasks(index))
```

Comment:
when the numFailures is bigger than maxTaskFailures, not need to append new a task and submit it